### ISSUE TITLE
Changing gradient transforms doesn't re-render the widget

### ISSUE THREAD

**Issue by everskies** [flutter/flutter#91952]
Whilst working on a customizable widget, I've discovered that changing the transform value for a gradient (used for a BoxDecoration) doesn't update the widget. 

**How to reproduce:**
https://dartpad.dev/58d8bc959528bf3cf35b413e77221e3a

The reason why this occurs is that the `bool operator ==` function inside LinearGradient and RadialGradient does not take the transform into account.

```dart
  @override
  bool operator ==(Object other) {
    if (identical(this, other))
      return true;
    if (other.runtimeType != runtimeType)
      return false;
    return other is LinearGradient
        && other.begin == begin
        && other.end == end
        && other.tileMode == tileMode
        && listEquals<Color>(other.colors, colors)
        && listEquals<double>(other.stops, stops);
```

**Expected results:**
Changing the transform angle should update/re-render the widget

**Actual results:** 
Changing the transform angle does not update/re-render the widget

**Comment by Zabadam** [comment]
I used this workaround in a `RenderLimitedBox` that I am working on

```dart
set gradient(Gradient gradient) {
  if (_gradient == gradient && _gradient.transform == gradient.transform)
    return;
  _gradient = gradient;
  markNeedsPaint();
}
```

just before searching for this issue.

Does "waiting for PR to land (fixed)" mean that sometime soon, the default behavior will be something like the following? If so, when could that be?

```dart
return other is LinearGradient
        && other.begin == begin
        && other.end == end
        && other.tileMode == tileMode
        && other.transform == transform
        && listEquals<Color>(other.colors, colors)
        && listEquals<double>(other.stops, stops);

@override
int get hashCode => hashValues(begin, end, tileMode, transform, hashList(colors), hashList(stops));
```

**Comment by Zabadam** [comment]
> I used this workaround in a `RenderLimitedBox` that I am working on
> . . .

I forgot I made this reply three years ago when I discovered #149534 and proposed fix #149624.

Since then, and seeing this bot label notification, I also noticed the `FooGradient.scale()` subclass methods are dropping the transform as well. (Should open a new issue.)

**Comment by Piinks** [comment]
@Zabadam does that mean this was fixed by https://github.com/flutter/flutter/pull/149624?
